Scribe/MainEditorForm.cs

```diff
@@ -404,37 +406,58 @@ private void UpdateLibraryDataDisplay()
         /// <param name="e">Ignored.</param>
         private void ContentAlteredEventHandler(object sender, EventArgs e)
         {
+            // TODO Remove this debug statement.
+            MainToolStripStatusLabel.Text = ChangeManager.Count.ToString();
+
             if (sender is TextBox textbox
                 && string.Compare(textbox.Text,
                                   EditableControls[typeof(TextBox)][textbox],
                                   comparisonType: StringComparison.OrdinalIgnoreCase) != 0)
             {
-                EditableControls[typeof(TextBox)][textbox] = textbox.Text;
-                HasUnsavedChanges = true;
+                ChangeManager.AddAndExecute(new Change(EditableControls[typeof(TextBox)][textbox], textbox.Text, textbox.Name,
+                                            (object databaseValue) => { _ = databaseValue.ToString(); HasUnsavedChanges = true; },
+                                            (object displayValue) => textbox.Text = displayValue.ToString(),
+                                            (object oldValue) => EditableControls[typeof(TextBox)][textbox] = oldValue.ToString()));
+                
             }
             else if (sender is CheckBox checkbox
                      && string.Compare(checkbox.Checked.ToString(),
```

Comment:
This is a simple example of when having object instead of string for the dictionary values would come in to play. You could just compare their boolyness... I think? You may have to unbox the type still

Reply:
This is a great point and I actually ended up making this change myself later on.
Good call catching this one before I did.